pkg/performanceprofile/controller/performanceprofile/components/tuned/tuned_test.go

```diff
@@ -166,28 +168,67 @@ var _ = Describe("Tuned", func() {
+						RealTime:             pointer.BoolPtr(true),
 					}
 					tunedData := getTunedStructuredData(profile)
 					bootLoader, err := tunedData.GetSection("bootloader")
 					Expect(err).ToNot(HaveOccurred())
-					Expect(bootLoader.Key("cmdline_idle_poll").String()).ToNot(Equal(cmdlineIdlePoll))
+					Expect(bootLoader.Key("cmdline_idle_poll").String()).To(Equal(cmdlineIdlePoll))
 				})
 			})
 
 			When("realtime workload disabled", func() {
 				It("should contain idle=poll cmdline", func() {
 					profile.Spec.WorkloadHints = &performancev2.WorkloadHints{
 						HighPowerConsumption: pointer.BoolPtr(true),
-						RealTime:             pointer.BoolPtr(true),
+						RealTime:             pointer.BoolPtr(false),
 					}
 					tunedData := getTunedStructuredData(profile)
 					bootLoader, err := tunedData.GetSection("bootloader")
 					Expect(err).ToNot(HaveOccurred())
-					Expect(bootLoader.Key("cmdline_idle_poll").String()).To(Equal(cmdlineIdlePoll))
+					Expect(bootLoader.Key("cmdline_idle_poll").String()).ToNot(Equal(cmdlineIdlePoll))
 				})
 			})
+
+			When("perPodPowerManagement Hint to true", func() {
+				It("should fail as PerPodPowerManagement and HighPowerConsumption can not be set to true", func() {
+					profile.Spec.WorkloadHints = &performancev2.WorkloadHints{
+						HighPowerConsumption:  pointer.BoolPtr(true),
+						PerPodPowerManagement: pointer.BoolPtr(true),
+					}
+					_, err := NewNodePerformance(profile)
+					Expect(err).To(HaveOccurred())
+				})
+			})
+		})
+
+		When("perPodPowerManagement Hint is false", func() {
+			It("should not contain perPodPowerManagement related parameters", func() {
+				profile.Spec.WorkloadHints.PerPodPowerManagement = pointer.BoolPtr(false)
+				tunedData := getTunedStructuredData(profile)
+				cpuSection, err := tunedData.GetSection("cpu")
+				Expect(err).ToNot(HaveOccurred())
+				Expect(cpuSection.Key("enabled").String()).ToNot(Equal("false"))
+				bootLoaderSection, err := tunedData.GetSection("bootloader")
+				Expect(err).ToNot(HaveOccurred())
+				Expect(bootLoaderSection.Key("cmdline_pstate").String()).ToNot(Equal(cmdlineHighPowerConsumptionPstate))
+				Expect(bootLoaderSection.Key("cmdline_pstate").String()).ToNot(Equal(cmdlinePerPodPowerManagementHint))
+			})
+		})
+
+		When("perPodPowerManagement Hint to true", func() {
+			It("should contain perPodPowerManagement related parameters", func() {
+				profile.Spec.WorkloadHints.PerPodPowerManagement = pointer.BoolPtr(true)
+				tunedData := getTunedStructuredData(profile)
+				cpuSection, err := tunedData.GetSection("cpu")
+				Expect(err).ToNot(HaveOccurred())
+				Expect(cpuSection.Key("enabled").String()).To(Equal("false"))
+				bootLoader, err := tunedData.GetSection("bootloader")
+				Expect(err).ToNot(HaveOccurred())
+				Expect(bootLoader.Key("cmdline_pstate").String()).To(Equal(cmdlinePerPodPowerManagementHint))
+				Expect(bootLoader.Key("cmdline_pstate").String()).ToNot(Equal(cmdlineHighPowerConsumptionPstate))
```

Comment:
when `cmdline_pstate` should be equal to `cmdlineHighPowerConsumptionPstate`?

Reply:
Added in previous test when only highPowerConsumption is to true:

`			When("default realtime workload settings", func() {
				FIt("should contain high power consumption related parameters", func() {
					profile.Spec.WorkloadHints = &performancev2.WorkloadHints{HighPowerConsumption: pointer.BoolPtr(true)}
					tunedData := getTunedStructuredData(profile)
					bootLoader, err := tunedData.GetSection("bootloader")
					Expect(err).ToNot(HaveOccurred())
					Expect(bootLoader.Key("cmdline_power_performance").String()).To(Equal(cmdlineHighPowerConsumption))
					Expect(bootLoader.Key("cmdline_pstate").String()).To(Equal(cmdlineHighPowerConsumptionPstate))
				})
			})`